src/cargo/ops/resolve.rs

```diff
@@ -304,8 +303,14 @@ pub fn resolve_with_previous<'gctx>(
         version_prefs.version_ordering(VersionOrdering::MinimumVersionsFirst)
     }
     if ws.resolve_honors_rust_version() {
-        let rust_version = ws.rust_version().cloned().map(RustVersion::into_partial);
-        version_prefs.max_rust_version(rust_version);
+        let rust_version = if let Some(ver) = ws.rust_version() {
+            ver.clone().into_partial()
+        } else {
+            let rustc = ws.gctx().load_global_rustc(None)?;
```

Comment:
Should we respect `.rustc_info.json`? There are fingerprint detection so it's pretty much safe if rustc has switched to other channel.

Reply:
That is done automatically through `load_global_rustc`

Reply:
I didn't get it. It seems that rustc info cache is disabled when no workspace provided. 

https://github.com/rust-lang/cargo/blob/7e31f62a80260c70b0e7604c615ed06fcf98928c/src/cargo/util/rustc.rs#L236-L246